include/FanFicArchive.py

```diff
@@ -0,0 +1,85 @@
+from bs4 import BeautifulSoup; from urllib import request; from re import match; import requests
+
+parser = "lxml"
+
+def fanfic_req(link): return BeautifulSoup(request.urlopen(link).read(), parser)
+def fanfic_check(id, name):
+    if isinstance(id, int): id = str(id)
+    else: assert isinstance(id, str), "Error for" + name + ": id not int or str"
+
+def ao3_rip(ao3_id): # downloads whole story in one go
+    fanfic_check(ao3_id, "ao3_rip")
+    text = fanfic_req('http://archiveofourown.org/works/'+ao3_id+'?view_adult=true&view_full_work=true')
+    tag_bar = text.find("dl", class_="stats").find_all("dd") # everything in the top bar
+    tag_list = {}
+    for i in tag_bar:
+        tag = [j.get_text() for j in i.find_all("a", class_="tag")]
+        if tag == []: tag = [i.get_text().lstrip('\n').lstrip(' ').rstrip(' ').rstrip('\n')]
+        tag_list[i.get("class")[0]] = tag
+    tag_list['author'] = [i.get_text() for i in text.find_all("a", rel="author")]
+    summary = text.find_all("div", class_="summary module", role="complementary") # just below top bar
+    chapter = {}
+    for i in text.find_all("div", class_="chapter", id=True): # separating each chapters
+        x = i.find("h3", class_="title").find("a").get("href").split("/")[-1]
+        chapter[i.get('id')] = [x, i]
+    return ao3_id, tag_list, summary, chapter
+
+def ff_rip(ff_id): # no image
+    fanfic_check(ff_id, "ff_rip")
+    text = fanfic_req("https://www.fanfiction.net/s/" +ff_id +"/1") # first chapter
+    tag_line = text.find("span", class_="xgray xcontrast_txt").get_text().split(" - ") # top bar
+    tag_list = {}
+    for i in tag_line:
+        if ":" in i: bracket, item = i.split(": ")[0:2]; tag_list[bracket] = item.rstrip(' ')
```

Comment:
Perhaps you wanted `i.split(": ", 1)` instead of `i.split(": ")[0:2]`?

Reply:
Dunno that existed